### PR TITLE
Added Mobius Function

### PR DESCRIPTION
**Pull request by QuantumNovice**

### CODE REVIEW

File: maths/is_square_free.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    '''
+    This functions takes a list of prime factors as input.
+    returns True if the factors are square free.
+
+    >>> is_square_free([1,1,2,3,4])
+    False
+
+    These are wrong but should return some value
+    it simply checks for repition in the numb
+    >>> is_square_free([1,3,4,'sd', 0.0])
```
Comment (cclauss) [inline maths/is_square_free.py]: `<space> between list items`

File: maths/is_square_free.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    True
+    
+    >>> is_square_free([1,0.5, 2, 0.0])
```
Comment (cclauss) [inline maths/is_square_free.py]: space

File: maths/is_square_free.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    True
+    
+    >>> is_square_free([1,2, 2, 5])
```
Comment (cclauss) [inline maths/is_square_free.py]: space

File: maths/is_square_free.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+'''
+References: wikipedia:square free number
+'''
+from typing import List
+
+def is_square_free(factors:List[int]) -> bool:
```
Comment (cclauss) [inline maths/is_square_free.py]: Please run this code through __python/black__.

File: maths/is_square_free.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    
+    >>> is_square_free('asd')
+    True
+    
+    >>> is_square_free(24)
+    Traceback (most recent call last):
+        ...
+    TypeError: 'int' object is not iterable
+    '''
+    i: int
```
Comment (cclauss) [inline maths/is_square_free.py]: i is an unused variable.

File: maths/is_square_free.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    if len(set(factors)) != len(factors):
+            return False
+    return True
```
Comment (cclauss) [inline maths/is_square_free.py]: Three lines --> one.
`    return len(set(factors)) == len(factors)`

File: maths/mobius_function.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    if n > 1:
+        factors.append(n)
+    return factors
+
+def is_square_free(factors:List[int]) -> bool:
+    '''
+    This functions takes a list of prime factors as input.
+    returns True if the factors are square free.
+
+    >>> is_square_free([1,1,2,3,4])
```
Comment (cclauss) [inline maths/mobius_function.py]: space

File: maths/mobius_function.py [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    False
+
+    These are wrong but should return some value
+    it simply checks for repition in the numb
+    >>> is_square_free([1,3,4,'sd', 0.0])
```
Comment (cclauss) [inline maths/mobius_function.py]: space

File: maths/mobius_function.py [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    True
+    
+    >>> is_square_free([1,0.5, 2, 0.0])
```
Comment (cclauss) [inline maths/mobius_function.py]: space

File: maths/mobius_function.py [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    True
+    
+    >>> is_square_free([1,2, 2, 5])
```
Comment (cclauss) [inline maths/mobius_function.py]: space

File: maths/mobius_function.py [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    Traceback (most recent call last):
+        ...
+    TypeError: '<=' not supported between instances of 'int' and 'list'
+    '''
+    factors = prime_factors(n)
+    if is_square_free(factors):
+        if len(factors)%2 == 0:
+            return 1
+        elif len(factors)%2 != 0:
+            return -1
```
Comment (cclauss) [inline maths/mobius_function.py]: Four lines --> one line:
`    return -1 if len(factors) % 2 else 1`

File: maths/is_square_free.py (hunk 1 above)
Comment (QuantumNovice) [reply maths/is_square_free.py]: When I use `from prime_factors import prime_factors` the build fails. Why is that?

File: maths/is_square_free.py (hunk 5 above)
Comment (QuantumNovice) [reply maths/is_square_free.py]: > 
> 
> Code reuse and formatting...

I followed your instructions.
I uploaded my old code without doing a proper anlysis, Sorry. 
Thank you! <3

File: maths/is_square_free.py (hunk 1 above)
Comment (cclauss) [reply maths/is_square_free.py]: Put a . (dot) as in __from .prime_factors import prime_factors__.

File: mobius_function.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
+"""
+Refrences: https://en.wikipedia.org/wiki/M%C3%B6bius_function
+References: wikipedia:square free number
+python/black : True
+flake8 : True
+"""
+
+from typing import List
+from .prime_factors import prime_factors
+from .is_square_free import is_square_free
```
Comment (cclauss) [inline mobius_function.py]: from maths..prime_factors import prime_factors
from maths.is_square_free import is_square_free

**Comment by cclauss** [comment]
Add an EMPTY file named `math/__init__.py` which will make __math__ a _package_.

**Comment by QuantumNovice** [comment]
Travis builds failed because they it ran on the old defective code (2nd Last commit). How do I trigger a new build?

**Comment by cclauss** [comment]
I restarted at https://travis-ci.org/TheAlgorithms/Python/builds/561450460?utm_source=github_status&utm_medium=notification

If you want to rebuild, you need to make and commit an insignificant change like add or modify a comment.  Just mark that commit with "Force a build" so it is clear to your readers.

File: maths/mobius_function.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
+"""
+Refrences: https://en.wikipedia.org/wiki/M%C3%B6bius_function
+References: wikipedia:square free number
+python/black : True
+flake8 : True
```
Comment (cclauss) [inline maths/mobius_function.py]: Did you run __flake8 maths/mobius_function.py__ on your local machine and did it pass?  If not then please change this line to __flake8 : --select=E9,F63,F7,F82__ to match https://github.com/TheAlgorithms/Python/blob/master/.travis.yml#L9

File: mobius_function.py [hunk 14]
```diff
@@ -0,0 +1,43 @@
+"""
```
Comment (cclauss) [inline mobius_function.py]: You have two files: __mobius_function.py__ and __maths/mobius_function.py__.  This file needs to be removed

**Comment by cclauss** [comment]
I think we are really close here. I was able to get the tests to pass locally just by doing `touch maths/__init__.py__` and `git add __init__.py__`.  This makes __maths__ into a package which allows the imports to work.  The one downside of this is that when you run locally you need to run from in the directory __above__ maths.  So you __maths/mobius_function.py__ to run that script and __code mobius_function.py__ to edit that script, etc.

File: maths/mobius_function.py (hunk 13 above)
Comment (cclauss) [reply maths/mobius_function.py]: $ flake8 .
```
./maths/mobius_function.py:8:1: F401 'typing.List' imported but unused
./maths/mobius_function.py:12:1: E302 expected 2 blank lines, found 1
./maths/mobius_function.py:39:1: E305 expected 2 blank lines after class or function definition, found 1
```
__python/black__ will autofix two of these.

File: maths/mobius_function.py [hunk 15]
```diff
@@ -0,0 +1,42 @@
+"""
+Refrences: https://en.wikipedia.org/wiki/M%C3%B6bius_function
+References: wikipedia:square free number
+python/black : True
+flake8 : True
+"""
+
+from typing import List
```
Comment (cclauss) [inline maths/mobius_function.py]: ./maths/mobius_function.py:8:1: F401 'typing.List' imported but unused

File: maths/mobius_function.py (hunk 13 above)
Comment (QuantumNovice) [reply maths/mobius_function.py]: I fixed it. I ran flake 8 on the original code.

File: maths/mobius_function.py (hunk 15 above)
Comment (QuantumNovice) [reply maths/mobius_function.py]: I fixed it. Flake 8 passed.

**Comment by cclauss** [comment]
Merged.

**Pull request merged**